userapi/src/main/java/controller/RouterVerticle.java

```diff
@@ -0,0 +1,148 @@
+package controller;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.json.Json;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.TimeoutHandler;
+import model.D42Model;
+import model.IDataModel;
+import model.Profile;
+
+import java.util.Set;
+
+/**
+ * Routes requests to their respective handlers
+ * Created by rohit.patiyal on 18/01/17.
+ */
+public class RouterVerticle extends AbstractVerticle {
+
+    private static final long REQ_TIMEOUT = 2000;
+    private IDataModel dataModel = null;
+
+    private void setIDataModel(IDataModel IDataModel) {
+        if (dataModel == null) {
+            this.dataModel = IDataModel;
+        }
+    }
+
+    private Router configureRouter() {
+        Router router = Router.router(vertx);
+        router.route().handler(TimeoutHandler.create(REQ_TIMEOUT));
+        router.route("/").handler(routingContext -> routingContext.response()
+                .putHeader("context-type", "text/html")
+                .end("<h1>Welcome to UserAPI for FKProfiler"));
+        router.get("/apps").blockingHandler(this::getAppIds, false);
```

Comment:
Will make the necessary change of using the default handler instead of blocking handler. I, previously, thought that the change was required just to make the list API async.